core/worker.go

```diff
@@ -937,9 +946,11 @@ func (w *worker) prepareWork(genParams *generateParams, block *types.Block) (*en
 				}
```

Comment:
env.uncles needs to be RLock() in commitUncles

Reply:
added a rLock